compiler/rustc_borrowck/src/type_check/input_output.rs

```diff
@@ -207,4 +200,27 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
             }
         }
     }
+
+    pub(crate) fn normalize_and_add_constraints(&mut self, t: Ty<'tcx>) -> Fallible<Ty<'tcx>> {
+        let TypeOpOutput { output: norm_ty, constraints, .. } =
```

Comment:
I forget, what are we not destructoring? For a bit, I thought we were ignoring obligations, but then I realized we switched to using a TypeOp (good idea).

Reply:
`canonicalized_query: Option<Canonical<'tcx, Op>>`, which is only used for diagnostics purposes.